src/common/page/Page.tsx

```diff
@@ -0,0 +1,65 @@
+import { PropsWithChildren, useEffect, useState } from 'react';
+
+import { Flex } from '@chakra-ui/react';
+import { Helmet } from 'react-helmet';
+import { useLocation, useNavigate, useParams } from 'react-router';
+
+import { Term } from 'lib/fetchers';
+import { useSessionStorage } from 'lib/hooks/storage/useSessionStorage';
+import { getCurrentTerm } from 'lib/utils/terms';
+
+import { Header } from 'common/header';
+import { ContentSidebar } from 'common/sidebar';
+
+type Props = {
+  title?: string;
+  hasSidebar?: boolean;
+  mobileSupport?: boolean;
+};
+
+export function Page({ title, hasSidebar, children }: PropsWithChildren<Props>) {
+  const [query, setQuery] = useState('');
+  const [savedTerm, setSavedTerm] = useSessionStorage('user:term', getCurrentTerm());
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const { term } = useParams();
+
+  const route = location.pathname.split('/')[1];
+
+  const handleSearchChange = (q: string) => {
+    if (!hasSidebar) {
+      navigate(`/calendar/${savedTerm}`);
+    }
+    setQuery(q);
+  };
+
+  useEffect(() => {
+    if (term) {
+      setSavedTerm(term);
+    } else if (route) {
+      navigate(`/${route}/${savedTerm}`, { replace: true });
+    }
+  }, [navigate, route, savedTerm, setSavedTerm, term]);
+
+  return (
+    <Flex h="100vh" direction="column" overflowX="hidden" overflowY="hidden">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <Header onSearchChange={handleSearchChange} />
+      {hasSidebar ? (
+        <Flex grow={1} overflow="hidden">
+          <ContentSidebar term={term as Term} searchQuery={query} />
```

Comment:
I have a feeling that we shouldn't make the sidebar with a page by default a search one? This is the "SearchSidebar". It's not a generic sidebar container. The developer won't be able to configure this page component to use some other content within the sidebar area and they'll make a new component. 

Page > PageWithSidebar
PageWithSidebar > PageWithCoursesSidebar > Scheduler 
PageWithSidebar > TimetableSharing

Reply:
If we can make components like the SimpleGrid components from Chakra or the Modal components, it might help keep the code more composable and reusable.